index.js

```diff
@@ -227,6 +236,10 @@ let rcl = {
         throw new TypeError('argv elements must not be null');
       }
 
+      // setup internal state for context outside promise. 
+      rclnodejs.init(context.handle(), argv);
+      this._nodes.set(context, []);
+
       let that = this;
       if (!this._initialized) {
```

Comment:
If we are using the existence of a context, which is used to init this module, to judge if the module has initialized, I think we could remove |_initialized| flag now.

Reply:
I think a flag for indicating that the ros version has been checked (new idl files compiled when changed) has value. 
What do you think about renaming `_initialized` to `_rosVersionChecked` and using it in a slightly simplified init() function?

tl,dr; My initial thought was that we could replace `_initialized` with a  check for _contextToNodeArrayMap.size > 0 but this would not address an edge case where:
```
let context = new Context();
await rclnodejs.init(context);
rclnodejs.shutdown(context);
let newContext = new Context();
await rclnodejs.init(newContext); // a new version check would be conducted 
```
While very unlikely, I opted to retain and rename `_initialized` to `_rosVersionChecked`. Push back if you disagree.